File: combinatorics/resource/ST/project_5/bai_toan_4/general_graph/al_to_eal.cpp
#include <iostream>
#include <vector>
#include <tuple>
#include <sstream>

using namespace std;

// Ham chuyen danh sach ke sang danh sach ke mo rong
vector<tuple<int, int, int>> alToEal(const vector<vector<int>>& adjList) {
    vector<tuple<int, int, int>> eal;
    for (int u = 0; u < adjList.size(); ++u) {
        for (int v : adjList[u]) {
            eal.emplace_back(u, v, 1); // Gan trong so la 1
        }
    }
    return eal;
}

int main() {
    int n;
    cout << "Nhap so dinh: ";
    cin >> n;
    cin.ignore(); // Loai bo ky tu xuong dong sau khi nhap n

    vector<vector<int>> adjList(n);
    cout << "Nhap danh sach ke (AL):\n";
    for (int i = 0; i < n; ++i) {
        cout << i << ": ";
        string line;
        getline(cin, line);
        stringstream ss(line);
        int v;
        while (ss >> v)
            adjList[i].push_back(v);
    }

    auto eal = alToEal(adjList);

    cout << "Danh sach ke mo rong (EAL):\n";
    for (const auto& [u, v, w] : eal)
        cout << u << " " << v << " " << w << "\n";

    return 0;
}

File: combinatorics/resource/ST/project_5/bai_toan_4/general_graph/al_to_eal.py
def al_to_eal(adj_list):
    eal = []
    for u in range(len(adj_list)):
        for v in adj_list[u]:
            eal.append((u, v, 1))  # Them canh (u, v) voi trong so 1
    return eal

# Nhap tu ban phim
n = int(input("Nhap so dinh: "))
adj_list = []
print("Nhap danh sach ke (AL):")
for i in range(n):
    row = list(map(int, input(f"{i}: ").split()))
    adj_list.append(row)

eal = al_to_eal(adj_list)

print("Danh sach ke mo rong (EAL):")
for u, v, w in eal:
    print(f"{u} {v} {w}")